src/transformers/generation/utils.py

```diff
@@ -339,6 +342,55 @@ class GenerateBeamEncoderDecoderOutput(ModelOutput):
 GenerateOutput = Union[GenerateNonBeamOutput, GenerateBeamOutput]
 
 
+class AssistantVocabTranslatorCache:
```

Comment:
Let's move this to `candidate_generator.py`, since this class is tightly coupled to `AssistantToTargetTranslator` (and to keep the main generation file, `utils.py`, with the bare minimum)